src/controllers/controllerengine.cpp

```diff
@@ -838,46 +821,55 @@ QScriptValue ControllerEngine::connectControl(QString group, QString name,
             proxy->disconnect();
         }
     } else {
-        qWarning() << "Invalid callback";
+        qWarning() << "Cannot make connection to" << group << name
+                   << ": no such callback function";
         return QScriptValue(false);
     }
 
-    if (function.isFunction()) {
-        qDebug() << "Connection:" << group << name;
+    QScriptContext *ctxt = m_pEngine->currentContext();
+    // Our current context is a function call to engine.connectControl. We
+    // want to grab the 'this' from the caller's context, so we walk up the
+    // stack.
+    if (ctxt) {
+        ctxt = ctxt->parentContext();
+        conn.context = ctxt ? ctxt->thisObject() : QScriptValue();
+    }
+    
+    if (!callback.isString()) {
```

Comment:
I would prefer to introduce an else branch, and copy the "conn.id =" from above, or integrate the body two times to the condition tree above. This way we have  the conn.id set operations in a single place.